### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by ykszk**
Adding windows support with the following changes:

- Add missing `typedef`s in the wrapper for msvc
- Add/Remove missing/extra libraries to link in windows
- Add CI test

### CODE REVIEW

**Comment by pevers** [comment]
Thanks for your work! This looks good to me. On a side note, maybe it would be good to transfer the repository to @Enet4 because I'm very busy at the moment with little time at my hands to maintain.

What do you think @Enet4 ?

**Comment by Enet4** [comment]
@pevers I can contribute to the maintenance of gdcm-rs, whether through repository transfer or just by being added as a collaborator and crate owner. My primary focus will continue to be on DICOM-rs and giving it first-class support for various transfer syntaxes without depending on GDCM, but for as long as there is value in this integration, it is in my best interest to keep it working. :)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (Enet4) [inline .github/workflows/test.yml]: The latest major version is v4.

```suggestion
    - uses: actions/checkout@v4
```

**Review by Enet4 — COMMENTED** [review]
I was able to compile this on a Windows 11 machine, as well as run the tests of `dicom-pixeldata` at `dicom-rs` after replacing `gdcm-rs` with a file-system path containing this version.

There was a warning during the native library compilation step:

```none
warning: gdcm-rs@0.5.0: cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
```

It is possible that msvc does not understand this argument.

More comments inline.

**Comment by ykszk** [comment]
Thank you for the reviews! I updated build flags for cleaner builds.

**Comment by pevers** [comment]
> @pevers I can contribute to the maintenance of gdcm-rs, whether through repository transfer or just by being added as a collaborator and crate owner. My primary focus will continue to be on DICOM-rs and giving it first-class support for various transfer syntaxes without depending on GDCM, but for as long as there is value in this integration, it is in my best interest to keep it working. :)

Yes understandable! I'll add you as a collaborator and crate owner.

**Pull request merged**